### PR TITLE
Resolve GSSP metadata generation twig environment configuration issue

### PR DESCRIPTION
**Pull request by VadimSchmitz**
Work done in this pull request:

- The local service `second_factor_only.metadata_factory` has now been removed so it uses the `Stepup-saml-bundle` configuration from the Stepup Saml Bundle.
- Update `SurfnetStepupGatewaySamlStepupProviderExtension.php` to use the new rendering method `twig` instead of the no longer available `templating`
- Update `ResponseRenderingService.php`

https://www.pivotaltracker.com/story/show/181710432

### CODE REVIEW

**Review by MKodde — CHANGES_REQUESTED** [review]
Thanks for fixing the bug! I have some feedback:
1. You have introduced several formatting issues. That's what the PHP CS is triggering on I recon.
2. Please reword the first commit. Using the 'fix this and that' can always be rewritten to be more specific.
3. XML output looks great again!
4. I reworded your PR message and title slightly. Making it easier to read.

**Comment by VadimSchmitz** [comment]
The static-analysis tests currently pass, however the build currently fails due to a vulnerability warning.

File: src/Surfnet/StepupGateway/GatewayBundle/Service/ResponseRenderingService.php [hunk 1]
```diff
@@ -74,14 +74,14 @@ public function renderRequesterFailureResponse(ResponseContext $context)
...
-        return $this->renderSamlResponse(
+        return ($this->renderSamlResponse(
             $context,
             'unprocessable_response',
             $this->responseBuilder
                 ->createNewResponse($context)
                 ->setResponseStatus(Constants::STATUS_RESPONDER)
                 ->get()
-        );
+        ));
```
Comment (MKodde) [inline src/Surfnet/StepupGateway/GatewayBundle/Service/ResponseRenderingService.php:84]: Wrapping the return statement in `(` brackets `)` is not adding any value right?

File: src/Surfnet/StepupGateway/GatewayBundle/Service/ResponseRenderingService.php [hunk 2]
```diff
@@ -104,16 +104,18 @@ public function renderResponse(
      */
     private function renderSamlResponse(
         ResponseContext $context,
-        $view,
-        SAMLResponse $response
+        string          $view,
```
Comment (MKodde) [inline src/Surfnet/StepupGateway/GatewayBundle/Service/ResponseRenderingService.php:107]: :+1: for adding the type def!

**Review by MKodde — COMMENTED** [review]
One change should be reverted. The rest looks great now!

**Review by MKodde — APPROVED** [review]
Grade A job!

:ship:

**Pull request merged**